include/deal.II/base/type_traits.h

```diff
@@ -0,0 +1,98 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__type_traits_h
+#define dealii__type_traits_h
+
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/la_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+struct true_type_s {
+static const bool value = true;
+};
+
+/**
+* A true_type similar to C++.11 std::true_type
+* 
+* @author Uwe Köcher, 2017
+*/
+typedef true_type_s true_type;
```

Comment:
@tjhei c++11 has a `std::true_type` and `std::false_type` within `type_traits`, since c++11 is not fully supported by all compilers, I have written it in a c++11 similar style to avoid each time the line `{ static const bool value = true; }`